client/state/sites/domains/actions.js

```diff
@@ -0,0 +1,130 @@
+/**
+ * External dependencies
+ */
+import debugFactory from 'debug';
+import map from 'lodash/map';
+
+/**
+ * Internal dependencies
+ */
+import { createSiteDomainObject } from './assembler';
+import i18n from 'lib/mixins/i18n';
+import wpcom from 'lib/wp';
+import {
+	SITE_DOMAINS_RECEIVE,
+	SITE_DOMAINS_REQUEST,
+	SITE_DOMAINS_REQUEST_SUCCESS,
+	SITE_DOMAINS_REQUEST_FAILURE,
+} from 'state/action-types';
+
+/**
+ * Module vars
+ */
+const debug = debugFactory( 'calypso:site-domains:actions' );
+
+/**
+ * `action-creator`
+ * Returns an action object to be used in signalling that
+ * an object containing the domains for
+ * a given site have been received.
+ *
+ * @param {Number} siteId - identifier of the site
+ * @param {Object} response - WP REST-API domains endpoint response
+ * @returns {Object} the action object
+ */
+export const createDomainsReceiveAction = ( siteId, response ) => {
```

Comment:
I'm aware this name looks a little bit weird but TBH this function isn't _receiving domains_, it's just returning the action to be dispatched. That's reason of its name.
Maybe we can create other functions like the following ?
```es6
export const receiveDomains = ( site, response ) => {
  return dispatch => dispatch( createDomainsReceiveAction( siteId, response ) );
};
```

... and then 

```es6
receiveDomains( site, response );
```

Anyway I'm all ears to get the final decision.